addons/webhooks/clusterbootstrap_webhook.go

```diff
@@ -292,7 +292,11 @@ func (wh *ClusterBootstrap) getGVR(gk schema.GroupKind) (*schema.GroupVersionRes
 	}
 	apiResourceList, err := wh.cachedDiscoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, err
+		// Sometimes it is unable to retrieve the complete list of server APIs i.g. an aggregated API server is down,
+		// ErrGroupDiscoveryFailed is returned in this case. We can still get other API resources and ignore this error.
+		if _, ok := err.(*discovery.ErrGroupDiscoveryFailed); !ok {
+			return nil, err
+		}
 	}
```

Comment:
Please add unit test to cover this line.

Reply:
Point taken.